Source/Core/Core/AchievementManager.cpp

```diff
@@ -293,6 +298,247 @@ void AchievementManager::ActivateDeactivateRichPresence()
   INFO_LOG_FMT(ACHIEVEMENTS, "Rich presence (de)activated.");
 }
 
+void AchievementManager::FetchBadges()
+{
+  if (!m_is_runtime_initialized || !IsLoggedIn() || !Config::Get(Config::RA_BADGES_ENABLED))
+  {
+    if (m_update_callback)
+      m_update_callback();
+    return;
+  }
+  m_image_queue.Cancel();
+
+  if (m_player_badge.name != m_display_name)
+  {
+    m_image_queue.EmplaceItem([this] {
+      std::string name_to_fetch;
+      {
+        std::lock_guard lg{m_lock};
+        name_to_fetch = m_display_name;
+        if (name_to_fetch == m_player_badge.name)
+          return;
+      }
+      rc_api_fetch_image_request_t icon_request = {.image_name = name_to_fetch.c_str(),
+                                                   .image_type = RC_IMAGE_TYPE_USER};
+      Badge fetched_badge;
+      if (RequestImage(icon_request, &fetched_badge) == ResponseType::SUCCESS)
+      {
+        INFO_LOG_FMT(ACHIEVEMENTS, "Successfully downloaded player badge id {}.", name_to_fetch);
+        std::lock_guard lg{m_lock};
+        if (name_to_fetch != m_display_name)
+        {
+          WARN_LOG_FMT(ACHIEVEMENTS, "Requested outdated badge id {} for player id {}.",
+                       name_to_fetch, m_display_name);
```

Comment:
IMO this is an expected outcome due to the double-locking, so I don't think this should be a warning. I'd drop that to info, same for the other instances of this.

Reply:
Okay. I had just figured, it's definitely not an _error_, but it does seem to be a rare enough sequence of events that there might be a problem behind it, but info works for me.